xwiki-platform-core/xwiki-platform-repository/xwiki-platform-repository-server-api/src/main/java/org/xwiki/repository/internal/RepositoryManager.java

```diff
@@ -1102,4 +1198,38 @@ private boolean updateCollection(BaseObject extensionObject, String fieldName, C
 
         return needSave;
     }
+
+    public BaseObject getExtensionVersionObject(XWikiDocument extensionDocument, String version) {
+        return getExtensionVersionObject(extensionDocument, version, true);
+    }
+
+    public BaseObject getExtensionVersionObject(XWikiDocument extensionDocument, String version, boolean allowProxying)
+    {
+        if (version == null) {
+            List<BaseObject> objects =
+                    extensionDocument.getXObjects(XWikiRepositoryModel.EXTENSIONVERSION_CLASSREFERENCE);
+
+            if (objects == null || objects.isEmpty()) {
+                return null;
+            } else {
+                return objects.get(objects.size() - 1);
+            }
+        }
+
+        BaseObject extensionVersionObject =
+                extensionDocument.getObject(XWikiRepositoryModel.EXTENSIONVERSION_CLASSNAME, "version", version, false);
+
+        if( extensionVersionObject == null && allowProxying && isVersionProxyingEnabled(extensionDocument) ){
+             //no ExtensionVersionClass object so we need to create such object temporarily
```

Comment:
I applied modification to previously discussed code, that I save document after adding version object to it.
The case is - it's the easiest way, making use of existing and working code.
The biggest minus is that - after requesting previous version of extension, it data will be kept in object behind page until the extension is not updated. Maybe it's not such a big drawback? Especially, taking into consideration, that if one of previous version is requested once, it's quite high probability it will be requested again. 
WDYT?